### DISCUSSION TITLE
ClientSideCaching，出现两次clientId，导致第二次以及后续操作时，客户端缓存不刷新，是不是我打开方式不对哈

### DISCUSSION THREAD

**Discussion by xiatiandegaga** [2881099/FreeRedis#195] [Q&A]
![image](https://github.com/user-attachments/assets/bec31a42-7d22-4445-8700-e5540f69cf2d)
![image](https://github.com/user-attachments/assets/1fe4d353-11f4-40fc-b9d7-79059f045ab6)
![image](https://github.com/user-attachments/assets/b5ceabf3-ad2e-49fa-a758-bb13318ed89f)

**Comment by xiatiandegaga** [comment]
1.3.0版本freeredis

**Comment by xiatiandegaga** [comment]
![image](https://github.com/user-attachments/assets/b914ec80-3695-4c03-9580-899e7afc1682)
删除key后，并没有读取数据库并重新写入redis-server，而是直接读取了本地缓存，Not connected

**Comment by 2881099** [comment]
https://github.com/2881099/FreeRedis/blob/master/examples/console_net8_client_side_caching/Program.cs

这个项目测试一下

**Reply by xiatiandegaga** [reply]
![image](https://github.com/user-attachments/assets/d42a5210-b05b-46d5-88d1-e4500fd0122c)
有两个clientid，断点处等了3s，还是找的本地，会不会7.2.5redis有问题，我重启一个新redis看看

**Reply by 2881099** [reply]
你可以用测试程序，直接连目标服务器啊

**Reply by xiatiandegaga** [reply]
![image](https://github.com/user-attachments/assets/b13b9c5a-0b3e-4699-adae-4b0ae9d993cb)
![image](https://github.com/user-attachments/assets/19cb0707-4d42-4e84-9507-6c8831e7123e)
直接用测试程序连接服务器的redis的哈

**Reply by 2881099** [reply]
那就和你说的7.2.5没关系啊

**Reply by xiatiandegaga** [reply]
是的，目前有点懵

**Reply by xiatiandegaga** [reply]
搜了issue#178，我把protocol=RESP3改成了protocol=RESP2，就好了，虽然还是多个clientid，但是本地缓存删除了
